Whoops, our SONAME changes from https://github.com/NixOS/nix/pull/13995 are slightly broken for unstable versions.

```
       > clang++  -o libnixutil.2.3220250918_3a687eeb.dylib libnixutil.2.3220250918_3a687eeb.dylib.p/archive.cc.o libnixutil.2.3220250918_3a687eeb.dylib.p/args.cc.o libnixutil.2.3220250918_3a687eeb.dylib.p/base-n.cc.o libnixutil.2.3220250918_3a687eeb.d
       > ld: malformed 32-bit x.y.z version number: 2.3220250918_3a687eeb
       > clang++: error: linker command failed with exit code 1 (use -v to see invocation)
       > ninja: build stopped: subcommand failed.
```

--- a/pkgs/tools/package-management/nix/default.nix
+++ b/pkgs/tools/package-management/nix/default.nix
@@ -180,14 +180,14 @@ lib.makeExtensible (
       nix_2_31 = addTests "nix_2_31" self.nixComponents_2_31.nix-everything;
 
       nixComponents_git = nixDependencies.callPackage ./modular/packages.nix rec {
-        version = "2.31pre20250712_${lib.substring 0 8 src.rev}";
+        version = "2.32pre20250919_${lib.substring 0 8 src.rev}";
         inherit maintainers teams;
         otherSplices = generateSplicesForNixComponents "nixComponents_git";
         src = fetchFromGitHub {
           owner = "NixOS";
           repo = "nix";
-          rev = "b124512388378cd38c4e353ddb387905d296e877";
-          hash = "sha256-asBUtSonedNfMO0/Z6HUi8RK/y/7I1qBDHv2UryichA=";
+          rev = "07b96c1d14ab8695e5071fb73e19049fce8f3b6b";
+          hash = "sha256-9tR08zFwQ9JNohdfeb40wcLfRnicXpKrHF+FHFva/WA=";
         };
       };
 

--- a/pkgs/tools/package-management/nix/modular/tests/functional/package.nix
+++ b/pkgs/tools/package-management/nix/modular/tests/functional/package.nix
@@ -15,6 +15,7 @@
   nix-store,
   nix-expr,
   nix-cli,
+  toml11,
 
   busybox-sandbox-shell ? null,
 
@@ -70,6 +71,13 @@ mkMesonDerivation (
       ''
         cd $(readlink -e $PWD)
         echo $PWD | grep tests/functional
+      '';
+
+    # `toml11` upgrade causes these to fail in 2.32+: https://github.com/NixOS/nixpkgs/pull/442682
+    # Remove when that PR lands in master.
+    ${if lib.versionAtLeast (lib.versions.majorMinor version) "2.32" then "preCheck" else null} =
+      lib.optionalString (lib.versionOlder toml11.version "4.0") ''
+        rm -f ../lang/eval-fail-fromTOML-{over,under}flow*
       '';
 
     mesonCheckFlags = [

